experimental/packages/opentelemetry-sdk-metrics-base/test/aggregator/Histogram.test.ts

```diff
@@ -0,0 +1,185 @@
+    it('no exceptions', () => {
+      const aggregator = new HistogramAggregator([1, 10, 100]);
+      const prev = aggregator.createAccumulation();
+      prev.record(0);
+      prev.record(1);
+
+      const delta = aggregator.createAccumulation();
+      delta.record(2);
+      delta.record(11);
+
+      const expected = aggregator.createAccumulation();
+      // replay actions on prev
+      expected.record(0);
+      expected.record(1);
+      // replay actions on delta
+      expected.record(2);
+      expected.record(11);
+
+      assert.deepStrictEqual(aggregator.merge(prev, delta), expected);
+    });
+  });
+
+  describe('diff', () => {
+    it('no exceptions', () => {
+      const aggregator = new HistogramAggregator([1, 10, 100]);
+      const prev = aggregator.createAccumulation();
+      prev.record(0);
+      prev.record(1);
+
+      const curr = aggregator.createAccumulation();
+      // replay actions on prev
+      curr.record(0);
+      curr.record(1);
+      // perform new actions
+      curr.record(2);
+      curr.record(11);
+
+      const expected = new HistogramAccumulation([1, 10, 100], {
+        buckets: {
+          boundaries: [1, 10, 100],
+          counts: [0, 1, 1, 0],
+        },
+        count: 2,
+        sum: 13,
+      });
+
+      assert.deepStrictEqual(aggregator.diff(prev, curr), expected);
+    });
+  });
+
+  describe('toMetricData', () => {
+    it('transform with AggregationTemporality.DELTA', () => {
+      const aggregator = new HistogramAggregator([1, 10, 100]);
+
+      const accumulation = aggregator.createAccumulation();
+      accumulation.record(0);
+      accumulation.record(1);
+
+      const sdkStartTime: HrTime = [0, 0];
+      const lastCollectionTime: HrTime = [1, 1];
+      const collectionTime: HrTime = [2, 2];
+
+      const expected: MetricData = {
+        resource: defaultResource,
+        instrumentationLibrary: defaultInstrumentationLibrary,
+        instrumentDescriptor: defaultInstrumentDescriptor,
+        pointDataType: PointDataType.HISTOGRAM,
+        pointData: [
+          {
+            attributes: {},
+            startTime: lastCollectionTime,
+            endTime: collectionTime,
+            point: {
+              buckets: {
+                boundaries: [1, 10, 100],
+                counts: [1, 1, 0, 0],
+              },
+              count: 2,
+              sum: 1,
+            },
+          },
+        ],
+      };
+      assert.deepStrictEqual(aggregator.toMetricData(
+        defaultResource,
+        defaultInstrumentationLibrary,
+        defaultInstrumentDescriptor,
+        [[{}, accumulation]],
+        AggregationTemporality.DELTA,
+        sdkStartTime,
+        lastCollectionTime,
+        collectionTime,
+      ), expected);
+    });
+
+    it('transform with AggregationTemporality.CUMULATIVE', () => {
+      const aggregator = new HistogramAggregator([1, 10, 100]);
+
+      const accumulation = aggregator.createAccumulation();
```

Comment:
Same as above, here we check if the first accumulation was successful, but not that the accumulation is kept for the next export interval, right?

Reply:
Can be resolved as well.